src/scroll/systems/bind.rs

```diff
@@ -0,0 +1,107 @@
+use bevy::prelude::*;
+
+use crate::scroll::components::*;
+
+pub fn scroll_content_that_controls_scroll_handles(
+    changed_scroll_content_query: Query<
+        (&ScrollContent, &Node, &ControlledByScrollBars, &Parent),
+        Or<(Changed<ScrollContent>, Changed<Node>)>,
+    >,
+    scroll_bars_query: Query<(&ScrollBar, &Node)>,
+    nodes_query: Query<&Node>,
+    mut scroll_handles_query: Query<&mut ScrollHandle>,
+) {
+    for (scroll_content, scroll_content_node, controllers, scroll_parent) in
+        changed_scroll_content_query.iter()
+    {
+        for scroll_bar_entity in controllers.scroll_bars.iter() {
+            let (scroll_bar, scroll_bar_node) = scroll_bars_query
+                    .get(*scroll_bar_entity)
+                    .expect("ControlledByScrollBars.scroll_bars should have contained Entities with ScrollBar components.");
+
+            let mut _scroll_handle = scroll_handles_query
+                    .get_mut(scroll_bar.scroll_handle)
+                    .expect("ScrollBar.scroll_handle should have contained a reference to an Entity with a ScrollHandle component.");
+            let mut scroll_handle = _scroll_handle.bypass_change_detection();
+
+            let root_node = nodes_query
+                .get(scroll_parent.get())
+                .expect("ScrollContent should not be orphaned in the UI hierarchy.");
```

Comment:
This is the best example of this, but I'm enforcing "this Entity must have this Component" constraints like this everywhere. Can this not be captured in the type system somehow? I realize `Entity<With<Component>>` is probably not a practical direction, but it just feels out of place to `unwrap()` everything. 

As the author it's just annoying boilerplate, but if this were a crate, it would be a bad experience to have a bunch of panics because the user "didn't set it up right" - you shouldn't be ABLE to set it up wrong!